### PR TITLE
Remove unused attachment placeholder code

### PR DESCRIPTION
**Pull request by jakub-trzebiatowski**
### Details

I found this when working on https://github.com/Expensify/App/issues/25415. Discussed [here](https://github.com/Expensify/App/pull/25524#discussion_r1299182331) in a draft PR which I'd like to rebase on this one.

This code path seems to never trigger, as the report action's `isLoading` property seems to never be set.

Open questions:
- Is the property supposed to be set by the backend?
- Is this code _supposed_ to execute or is it some leftover?

Removing this will simplify fixing https://github.com/Expensify/App/issues/25415.

### Fixed Issues

Helps with https://github.com/Expensify/App/issues/25415

### Tests

- Go to a report with attachment messages
- Upload an attachment
  - [ ] Verify that it uploads correctly
  - [ ] Verify that the behavior is the same as on production
- Test interactions on the uploaded attachment
  - [ ] Verify that reactions, download, reply in thread and delete options work as expected
- Upload a [corrupted attachment](https://drive.google.com/file/d/1iOhC-C5CFh6x0W_a5ByTbKX8vW6tLTGW/view?usp=sharing)
  - [ ] Verify that the upload fails
  - [ ] Verify that the comment is dimmed

### QA Steps

Same as Tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/2590174/b99763b6-bd0f-464d-9811-7941c9156c47

https://github.com/Expensify/App/assets/2590174/f275423a-3f7c-48ba-9543-1fac344bb97a

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by flodnv** [comment]
>Is the property supposed to be set by the backend?

I don't see any instance of `isLoading` in our backend code, so I guess not. 

>Is this code supposed to execute or is it some leftover?

No idea honestly, but if it's unreachable I think it's pretty safe to just remove it?

**Review by flodnv — COMMENTED** [review]
>Is the property supposed to be set by the backend?

I don't see any instance of `isLoading` in our backend code, so I guess not. 

>Is this code supposed to execute or is it some leftover?

No idea honestly, but if it's unreachable I think it's pretty safe to just remove it?

**Review by flodnv — DISMISSED** [review]
LGTM - @marcaaron let us know your thoughts 🙇

**Comment by marcaaron** [comment]
yes, it is a relic! replaced with the `pendingAction` in this PR -> https://github.com/Expensify/App/pull/10317

I would assume based on the age of that PR that anything referencing `action.isLoading` is dead code.

**Comment by marcaaron** [comment]
Is there some linked issue though to go with this? What prompted this change?

**Comment by cubuspl42** [comment]
@marcaaron 

> Is there some linked issue though to go with this? What prompted this change?

It started with [this issue](https://github.com/Expensify/App/issues/24216) about some inconsistent 8px margin. I was the C+ reviewer. We introduced [a regression](https://github.com/Expensify/App/issues/25415) trying to fix that inconsistency. I concluded that the reason why it's so difficult to apply that simple margin properly is the unnecessary complexity of the surrounding code. So now I'm cleaning it up, then I'll rebase https://github.com/Expensify/App/pull/25524 on top of those cleanups. I hope it makes sense.

File: src/styles/StyleUtils.js
Comment (flodnv) [inline src/styles/StyleUtils.js:640]: I'm not sure I understand this change? `isLoading` makes sense, no?

File: src/pages/home/report/ReportActionItem.js
Comment (cubuspl42) [reply src/pages/home/report/ReportActionItem.js:559]: Soon we'll figure out that half of our code is dead 😶 I really hope that TypeScript will help us out here

Comment (cubuspl42) [reply src/pages/home/report/ReportActionItem.js:559]: I'll take a look at this

File: src/styles/StyleUtils.js
Comment (cubuspl42) [reply src/styles/StyleUtils.js:640]: Yes, I changed the name because [we are nuking](https://github.com/Expensify/App/pull/25557#discussion_r1299839956) `action.isLoading`, and this use had to be adjusted.

File: src/pages/home/report/ReportActionItem.js
Comment (flodnv) [reply src/pages/home/report/ReportActionItem.js]: Sorry, I'm not sure what exactly is the question?

Comment (cubuspl42) [reply src/pages/home/report/ReportActionItem.js]: Oh, it's just a heads-up for you / the internal team that there might be one more dead code path.

Comment (cubuspl42) [reply src/pages/home/report/ReportActionItem.js]: You can do whatever with this information

**Comment by marcaaron** [comment]
>It started with https://github.com/Expensify/App/issues/24216 about some inconsistent 8px margin. I was the C+ reviewer. We introduced https://github.com/Expensify/App/issues/25415 trying to fix that inconsistency. I concluded that the reason why it's so difficult to apply that simple margin properly is the unnecessary complexity of the surrounding code. So now I'm cleaning it up, then I'll rebase https://github.com/Expensify/App/pull/25524 on top of those cleanups. I hope it makes sense.

I don't want to discourage your efforts (which we are appreciative of and seems like this is necessary cleanup). But I think it doesn't _really_ make sense - or at least is unusual or outside of our normal process. Typically, if you want to do a refactor then you would propose somewhere the scope of the work. Then we create an issue for it. Then we accept that proposal. And then a PR like this get created and be linked to the issue. Then you will get compensated for it. Let me know if I am missing something.

**Comment by cubuspl42** [comment]
> But I think it doesn't really make sense - or at least is unusual or outside of our normal process.

I think it does make sense, but might indeed be outside of the process.

> Then you will get compensated for it. Let me know if I am missing something.

I don't expect to be compensated for these cleanup PRs. And I skipped the formalities because the gain of removing the dead code and obvious duplication is hardly debatable.

As I understand, per the process C/C+ are obligated to fix the regression if it occurs during the regression period. Do you confirm that? So the only thing which is outside the process is that I classified the root cause of the regression lying in the unnecessary code complexity and organized the fix into a few smaller PRs, instead of 1 big one.

**Comment by marcaaron** [comment]
>I think it does make sense, but might indeed be outside of the process.

Yes, the **change** makes sense - sorry the **explanation** is what did not make sense to me because it looked (to me) like we did not quite follow process. To be sure, removing the code and all that is a good thing.

Since this is part of the solution to a _regression_ then my recommendation would be to just link to the original issue in the description of this PR (so everyone can understand the context for the change). If you have made some plan to do the cleanup and got the approval to proceed then link to where that conversation happened.

To clarify my initial concern, it's that you might have been doing some random cleanup for which you would not be compensated for and nobody asked for. I see now that is a misunderstanding and that @Li357 gave you a green light already to open this PR.

**Comment by cubuspl42** [comment]
> it's that you might have been doing some random cleanup for which you would not be compensated for and nobody asked for

I still cannot see a problem here... I know that in general, it might be a challenge to accept PRs from the whole community because it's a cost to review them and might have negative net gain, but I think it's a bit different situation with the C+ group, who are filtered and have more experience in the project. We are also in the Expensify GitHub organization if that means anything. 

If during a review I accidentally see some dead code (let's say in a file not strictly related to the solved issue), what am I supposed to do?

> Typically, if you want to do a refactor then you would propose somewhere the scope of the work. Then we create an issue for it. Then we accept that proposal. 

This? If that's the expectation, I will just pretend I didn't see that dead code, and so will others (or they already are doing that).

I'm not gonna argue about this, but I think that if anything, it's this specific part of the process that makes little sense.

But coming back to this PR, are we ready to push it forward?

**Comment by marcaaron** [comment]
Overall, thanks for the thoughts and feedback! ❤️ 

>it might be a challenge to accept PRs from the whole community because it's a cost to review them and might have negative net gain, but I think it's a bit different situation with the C+ group, who are filtered and have more experience in the project. We are also in the Expensify GitHub organization if that means anything.

This is not situation we have here though after reviewing the linked issue. So, perhaps we would like to restart this conversation in Slack with different context?

>If during a review I accidentally see some dead code (let's say in a file not strictly related to the solved issue), what am I supposed to do?

It happens rarely in my experience. But sounds worth discussing.

We have various initiatives (particularly urgent ones at the moment) and a small internal team so it unfortunately does not _always_ make sense to accept every improvement. But if we want to have some larger discussion about whether C+ team should feel empowered to submit improvement PRs without any connection to an issue then I would encourage it. 👍 

Internally, most of us submit PRs connected to issues, but in very rare cases we do not. We are also salaried. So, my concerns with what you are suggesting boil down to:

1. Everybody should be able to see _why_ we are doing the work.
2. People get compensated for the work they are doing.

>But coming back to this PR, are we ready to push it forward?

Only remaining problem I see is this @cubuspl42:

**Comment by cubuspl42** [comment]
@marcaaron I linked https://github.com/Expensify/App/issues/25415 as the "Fixed issue", although I don't fix it (yet), so it's fiction. I hope that satisfies the process, though. I don't have any idea what proposal I could link.

**Comment by marcaaron** [comment]
If you remove the `$` from the link then you can reference whatever you want - or do so somewhere else in the description. I am mainly trying to establish that this PR should have some kind of link to some context about why we are doing it.

**Comment by marcaaron** [comment]
>I don't have any idea what proposal I could link

Link to whatever conversation prompted you to make the change. Just add some context somewhere please.

**Comment by cubuspl42** [comment]
@marcaaron I updated "Details" and "Fixed issues" sections

**Comment by cubuspl42** [comment]
@marcaaron Let me know if that's what you meant

**Comment by marcaaron** [comment]
Thanks the context is exactly what I have been looking for! Thank you for adding it 🙇 

As per this message in Slack it seems we are currently [on a merge freeze until September 11th](https://expensify.slack.com/archives/C01GTK53T8Q/p1693347701566439). 

@flodnv Should we put a HOLD on this so that we can pick it up once the freeze is lifted?

Thanks for your patience @cubuspl42

**Comment by cubuspl42** [comment]
@marcaaron The merge freeze is over, so we can try to get this merged

**Review by marcaaron — APPROVED** [review]
Changes LGTM. Can we improve these test steps a bit?

>Go to a report with attachment messages
>Verify that various attachment interactions work as expected

e.g. say what is the expected behavior to verify, specifically ask to upload, etc.

File: src/pages/home/report/ReportActionItem.js
Comment (situchan) [inline src/pages/home/report/ReportActionItem.js:590]: There's also the case when action has error. When error, opacity should be 0.5.
@cubuspl42 please add this error case to Tests step as well.

Comment (cubuspl42) [reply src/pages/home/report/ReportActionItem.js:590]: For a few seconds I was really confused, but now I get it; I just blindly removed the `isLoading` parameter, when `isLoading` was abused when there was an error. Thanks for spotting this.

Comment (cubuspl42) [reply src/pages/home/report/ReportActionItem.js:590]: @situchan  Actually, maybe a silly question, but how to trigger an error so `action.error` is truthy?

Comment (situchan) [reply src/pages/home/report/ReportActionItem.js:590]: I am also not sure.
Please try repro step in https://github.com/Expensify/App/issues/27315.

If we're not clear, let's keep that code part. And only remove `isLoading` conditions.

Comment (cubuspl42) [reply src/pages/home/report/ReportActionItem.js:590]: @bernhardoj I had a feeling that I've discussed this before, but this PR is so old that I already forgot the details...

@situchan If our conclusion from the other discussion looks good for you, we can leave this as-is. I'll also use the repro steps you linked to double confirm.

Comment (cubuspl42) [reply src/pages/home/report/ReportActionItem.js:590]: image

(with the code as-is)

Comment (cubuspl42) [reply src/pages/home/report/ReportActionItem.js:596]: @situchan Please let me know if we can resolve this thread without code changes

Comment (situchan) [reply src/pages/home/report/ReportActionItem.js:596]: I think we can resolve

**Comment by cubuspl42** [comment]
@situchan Can we move this forward?

**Comment by situchan** [comment]
Server is down - https://expensify.slack.com/archives/C01GTK53T8Q/p1695314249132269
Will checklist once it's back

**Comment by flodnv** [comment]
Can we merge this @situchan ?

**Comment by situchan** [comment]
yes, we can. For safety, I wanna test based on latest codebase as 1415 commits are too behind
@cubuspl42 please pull main.

**Comment by cubuspl42** [comment]
@situchan Done

**Comment by flodnv** [comment]
Bump @situchan

**Comment by cubuspl42** [comment]
@situchan Also, you can always merge `main` locally; if there are no conflicts, the conclusion from the testing reflects more-or-less the app behavior after the feature branch gets merge _into_ `main`, unless there is a flaw in my logic

**Comment by situchan** [comment]
Job 2 is failing. So still pull main

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details open>
<summary>Web</summary>


https://github.com/Expensify/App/assets/108292595/a41aa68e-b3b0-43c4-94e4-d9c4470d4d49



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Review by situchan — APPROVED** [review]
Confirmed the behavior is exactly same as `main`

**Review by flodnv — APPROVED** [review]

**Pull request merged**
